### PR TITLE
Whole Game (working version)

### PR DESCRIPTION
**Pull request by nchengg**

### CODE REVIEW

File: src/gseproject/EXEplorer/urls.py [hunk 1]
```diff
@@ -16,4 +17,6 @@
     path('api/get_carbon_footprint_value/', views.get_carbon_footprint_value, name='get_carbon_footprint_value'),
     path('api/increment_carbon_footprint/', views.increment_carbon_footprint, name='increment_carbon_footprint'),
     path('scan/', views.scan_view, name='scan'),
+    path('start-pet-walk/', views.petDisappear, name='start-pet-walk'),
+    path('pet-check', views.petCheck, name = 'pet-check'),
```
Comment (Yout-bit) [inline src/gseproject/EXEplorer/urls.py:21]: Do these need their own urls? can the attached functions be triggered without a url?

File: src/gseproject/templates/game/game_home.html [hunk 2]
```diff
@@ -6,8 +6,78 @@
...
+        } else {
+            petAppear();
+            // Handle non-blank response here
+        }
+        })
+            .catch((error) => {
+                console.error('There has been a problem with your fetch operation:', error);
+            });
+    }
+
```
Comment (Yout-bit) [inline src/gseproject/templates/game/game_home.html:86]: I feel these functions are generic enough the be in the template js file, check if there are already (might be under a different name)

File: src/gseproject/templates/game/shop.html [hunk 3]
```diff
@@ -9,67 +9,116 @@
...
+                console.log(data);
+                deductCherryCount();
+            })
+            .catch((error) => {
+                console.error('There has been a problem with your fetch operation:', error);
+            });
+    }
+    // Function to get CSRF token
+    function getCookie(name) {
+
```
Comment (Yout-bit) [inline src/gseproject/templates/game/shop.html]: **This is reapeated code. Remove it.**

**Review by Yout-bit — CHANGES_REQUESTED** [review]

File: src/gseproject/templates/game/scan.html [hunk 4]
```diff
@@ -9,5 +9,201 @@
...
+    </div>
+</div>
+
+<script>
+
+
+
+
+    // Predefined array of buildings with names and coordinates
+    const buildings = [
```
Comment (Yout-bit) [inline src/gseproject/templates/game/scan.html]: This needs to be a separate file

File: src/gseproject/templates/game/scan.html [hunk 5]
```diff
@@ -9,5 +9,201 @@
...
+                buildingsWithinRange.push(building);
+            }
+        });
+
+        return buildingsWithinRange;
+    }
+
+    // Function to handle scanning and checking location
+    // Function to handle scanning and checking location
+    function scanForCoordinate() {
```
Comment (Yout-bit) [inline src/gseproject/templates/game/scan.html]: If these functions are in here, they need to be removed from base_game_template.js

File: src/gseproject/EXEplorer/urls.py (hunk 1 above)
Comment (nchengg) [reply src/gseproject/EXEplorer/urls.py:21]: The attached functions cannot be triggered without a url. Its also API

File: src/gseproject/templates/game/game_home.html (hunk 2 above)
Comment (nchengg) [reply src/gseproject/templates/game/game_home.html:86]: .go button and .pet are all only in game_home.html

File: src/gseproject/templates/game/scan.html (hunk 4 above)
Comment (nchengg) [reply src/gseproject/templates/game/scan.html]: This file is from marco. Please deny this and check his pull

File: src/gseproject/templates/game/scan.html (hunk 5 above)
Comment (nchengg) [reply src/gseproject/templates/game/scan.html]: This file is from marco. Please deny this and check his pull

File: src/gseproject/templates/game/shop.html (hunk 3 above)
Comment (nchengg) [reply src/gseproject/templates/game/shop.html]: This a better version of indentation, syntax changes only

**Review by nchengg — COMMENTED** [review]
Please refer to the comments below

File: src/gseproject/templates/game/scan.html (hunk 4 above)
Comment (Yout-bit) [reply src/gseproject/templates/game/scan.html]: i cant deny individual files. plases claen up the whole request if you dont want it part of the request

File: src/gseproject/templates/game/shop.html (hunk 3 above)
Comment (Yout-bit) [reply src/gseproject/templates/game/shop.html]: Then it should be changed in it's orginal location

**Pull request merged**